### PR TITLE
feat: document tab improvement (FPLA-2767)

### PR DESCRIPTION
**Pull request by toireasam**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FPLA-2612

### Change description ###
Improvement to document tab

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/document/DocumentsListRendererTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.fpl.utils.ResourceReader.readString;
+
+@ExtendWith(MockitoExtension.class)
+class DocumentsListRendererTest {
```
Comment (samuelzerozero) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/document/DocumentsListRendererTest.java:19]: missing edge case with zero document views.

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/document/DocumentsListRendererTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    private final CaseUrlService caseUrlService = mock(CaseUrlService.class);
+
+    private final DocumentsListRenderer underTest = new DocumentsListRenderer(IMAGE_BASE_URL, caseUrlService);
+
+    @Test
+    void shouldRenderDocumentBundles() {
+        List<DocumentBundleView> documentBundleViews = List.of(
+            DocumentBundleView.builder()
+                .name("Applicant's statements and application documents")
```
Comment (samuelzerozero) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/document/DocumentsListRendererTest.java:32]: minor: could improve later: all the decision paths are covered, but they both included in 1 test only. Ideally there should be 3 separate tests, full/empty and multiple elements

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/document/DocumentsListRendererTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.fpl.utils.ResourceReader.readString;
+
+@ExtendWith(MockitoExtension.class)
+class DocumentsListRendererTest {
+
```
Comment (samuelzerozero) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/document/DocumentsListRendererTest.java:20]: missing edge case when getDocumentUrl() is incorrect and returns  an empty string

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/document/transformer/DocumentViewTestHelper.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            .party(RespondentParty.builder().firstName(firstName).lastName(lastName).build())
+            .build());
+    }
+
+    private static Element<SupportingEvidenceBundle> buildFurtherEvidenceBundle(
+        String name, String uploadedBy, boolean isConfidential, FurtherEvidenceType type) {
+        return element(SupportingEvidenceBundle.builder()
+            .name(name)
+            .document(testDocumentReference())
+            .dateTimeUploaded(LocalDateTime.now())
```
Comment (samuelzerozero) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/document/transformer/DocumentViewTestHelper.java:58]: can this give some issues? since is a now uncontrolled?

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/document/DocumentsListRendererTest.java (hunk 2 above)
Comment (samuelzerozero) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/document/DocumentsListRendererTest.java:32]: missing coverage for uploadedAt/uploadedBy

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/document/transformer/FurtherEvidenceDocumentsBundleTransformerTest.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+import static uk.gov.hmcts.reform.fpl.service.document.transformer.DocumentViewTestHelper.ADMIN_CONFIDENTIAL_DOCUMENT;
+import static uk.gov.hmcts.reform.fpl.service.document.transformer.DocumentViewTestHelper.ADMIN_NON_CONFIDENTIAL_DOCUMENT;
+import static uk.gov.hmcts.reform.fpl.service.document.transformer.DocumentViewTestHelper.LA_CONFIDENTIAL_DOCUMENT;
+import static uk.gov.hmcts.reform.fpl.service.document.transformer.DocumentViewTestHelper.LA_NON_CONFIDENTIAL_DOCUMENT;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.TIME_DATE;
+import static uk.gov.hmcts.reform.fpl.utils.DateFormatterHelper.formatLocalDateTimeBaseUsingFormat;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = {FurtherEvidenceDocumentsTransformer.class})
+class FurtherEvidenceDocumentsBundleTransformerTest {
```
Comment (samuelzerozero) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/document/transformer/FurtherEvidenceDocumentsBundleTransformerTest.java:28]: this class is not testing getFurtherEvidenceDocumentBundles and indirectly getFurtherEvidenceDocumentsView.

**Review by samuelzerozero — APPROVED** [review]
